dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/changelog/JdbcChangeLogStore.java

```diff
@@ -0,0 +1,97 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.changelog;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hisp.dhis.changelog.EventDataValueChangeLog;
+import org.hisp.dhis.changelog.EventDataValueChangeLog.Change;
+import org.hisp.dhis.program.UserInfoSnapshot;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Slf4j
+@Repository("org.hisp.dhis.tracker.export.changelog.ChangeLogStore")
+@RequiredArgsConstructor
+public class JdbcChangeLogStore implements ChangeLogStore {
+
+  private final JdbcTemplate jdbcTemplate;
+
+  private static final RowMapper<EventDataValueChangeLog> customEventChangeLogRowMapper =
+      (rs, rowNum) -> {
+        UserInfoSnapshot user = new UserInfoSnapshot();
+        user.setUsername(rs.getString("userName"));
+        user.setFirstName(rs.getString("firstname"));
+        user.setSurname(rs.getString("surname"));
+        user.setUid(rs.getString("useruid"));
+
+        return new EventDataValueChangeLog(
+            user,
+            rs.getTimestamp("updatedAt"),
+            new Change(
+                new EventDataValueChangeLog.DataValueChange(
+                    rs.getString("dataElementUid"),
+                    rs.getString("previousValue"),
+                    rs.getString("currentValue"))));
+      };
+
+  @Override
+  public List<EventDataValueChangeLog> getEventChangeLog(String eventUid) {
+    final String sql =
+        """
+            select
+              case
+                when cl.audittype = 'CREATE' and cl.currentchangelogvalue is null then cl.previouschangelogvalue
+                when cl.audittype = 'CREATE' and cl.currentchangelogvalue is not null then cl.previouschangelogvalue
```

Comment:
There are two cases for CREATE here with an `and` with the two possible values of `cl.previouschangelogvalue` leading to the same result. Can we simplify this to 

```suggestion
                when cl.audittype = 'CREATE' then cl.previouschangelogvalue
```